tests/functional/test_relationships.py

```diff
@@ -29,7 +29,7 @@ class EmployeeSchema(dy.Schema):
 
 @pytest.fixture()
 def departments() -> dy.LazyFrame[DepartmentSchema]:
-    return DepartmentSchema.cast(pl.LazyFrame({"department_id": [1, 2]}))
```

Comment:
AFAICT `departments` is always the `left` dataframe when we call `require_relationship_one_to_one` in the tests below. Don't we also need a test where the `left` dataframe is not unique in the join column?